lib/spack/spack/fetch_strategy.py

```diff
@@ -907,51 +926,51 @@ def __init__(self, root):
         self.root = os.path.abspath(root)
 
     def store(self, fetcher, relativeDst):
+        # skip fetchers that aren't cachable
         if not fetcher.cachable:
             return
 
+        # Don't store things that are already cached.
+        if isinstance(fetcher, CacheURLFetchStrategy):
```

Comment:
Ah fair enough. And this is clear anyhow. And now that I think about it having cachable return false is unclear. Anyhow this looks fine!